README.md

```diff
@@ -1,492 +1,271 @@
+{
+    private readonly IPactBuilderV3 PactBuilder;
 
-#### 3. Write your test
-Create a new test case and implement it.
+    public SomethingApiConsumerTests(ITestOutputHelper output)
+    {
+        // Use default pact directory ..\..\pacts and default log
+        // directory ..\..\logs
+        var pact = Pact.V3("Something API Consumer", "Something API", new PactConfig());
 
-```c#
-public class SomethingApiConsumerTests : IClassFixture<ConsumerMyApiPact>
-{
-  private IMockProviderService _mockProviderService;
-  private string _mockProviderServiceBaseUri;
-
-  public SomethingApiConsumerTests(ConsumerMyApiPact data)
-  {
-    _mockProviderService = data.MockProviderService;
-    _mockProviderService.ClearInteractions(); //NOTE: Clears any previously registered interactions before the test is run
-    _mockProviderServiceBaseUri = data.MockProviderServiceBaseUri;
-  }
-
-  [Fact]
-  public void GetSomething_WhenTheTesterSomethingExists_ReturnsTheSomething()
-  {
-    //Arrange
-    _mockProviderService
-      .Given("There is a something with id 'tester'")
-      .UponReceiving("A GET request to retrieve the something")
-      .With(new ProviderServiceRequest
-      {
-        Method = HttpVerb.Get,
-        Path = "/somethings/tester",
-        Headers = new Dictionary<string, object>
-        {
-          { "Accept", "application/json" }
-        }
-      })
-      .WillRespondWith(new ProviderServiceResponse
-      {
-        Status = 200,
-        Headers = new Dictionary<string, object>
+        // or specify custom log and pact directories
+        pact = Pact.V3("Something API Consumer", "Something API", new PactConfig
         {
-          { "Content-Type", "application/json; charset=utf-8" }
-        },
-        Body = new //NOTE: Note the case sensitivity here, the body will be serialised as per the casing defined
-        {
-          id = "tester",
-          firstName = "Totally",
-          lastName = "Awesome"
-        }
-      }); //NOTE: WillRespondWith call must come last as it will register the interaction
+            PactDir = @"..\pacts",
+            LogDir = @"c:\temp\logs",
+        });
 
-    var consumer = new SomethingApiClient(_mockProviderServiceBaseUri);
+        // Initialize Rust backend
+        PactBuilder = pact.UsingNativeBackend();
+    }
 
-    //Act
-    var result = consumer.GetSomething("tester");
+    [Fact]
+    public async Task GetSomething_WhenTheTesterSomethingExists_ReturnsTheSomething()
+    {
+        // Arrange
+        PactBuilder
+            .UponReceiving("A GET request to retrieve the something")
+                .Given("There is a something with id 'tester'")
+                .WithRequest(HttpMethod.Get, "/somethings/tester")
+                .WithHeader("Accept", "application/json")
+            .WillRespond()
+                .WithStatus(HttpStatusCode.OK)
+                .WithHeader("Content-Type", "application/json; charset=utf-8")
+                .WithJsonBody(new

```

Comment:
Typically we'd expect people to be using `Matcher` instead of a fixed object here wouldn't we? We don't document those anywhere else currently so it would be good to have the main example using a `Matcher` and explaining why that makes sense.